### ISSUE TITLE
false positive when creating new types

### ISSUE THREAD

**Issue by Johnrobmiller** [sweepline/eslint-plugin-unused-imports#60]
Compiler said these types are unused, which is absurd. Clearly a bug. Screenshot:
<img width="848" alt="image" src="https://github.com/sweepline/eslint-plugin-unused-imports/assets/55205360/cae52dc2-7160-45ec-82f6-27ae014614f6">

**Comment by sweepline** [comment]
What is your eslint config, are you using `@typescript-eslint`? This looks like youre not using the correct parser.

**Comment by JudithHeidi** [comment]
Can confirm same issue, both with function types as above and Enum types.

Using the built-in NextJS linting set-up plus a few extras, so overall the relevant & sanitised parts of my final config are:

```
  "plugins": [
    "react-hooks",
    "jsx-a11y",
    "react",
    "import",
    "@next/next",
    "@typescript-eslint",
    "@eslint-community/eslint-comments",
    "simple-import-sort",
    "unused-imports"
  ],
  "rules": {
    "no-unused-vars": [
      "off"
    ],
    "@typescript-eslint/no-unused-vars": [
      "off"
    ],
    "unused-imports/no-unused-imports": [
      "error"
    ],
    "unused-imports/no-unused-vars": [
      "warn",
      {
        "vars": "all",
        "varsIgnorePattern": "^_",
        "args": "after-used",
        "argsIgnorePattern": "^_"
      }
    ],
```
-stuff-
```
 "settings": {
    "react": {
      "version": "detect"
    },
    "import/parsers": {
      "<path>\\node_modules\\@typescript-eslint\\parser\\dist\\index.js": [
        ".ts",
        ".mts",
        ".cts",
        ".tsx",
        ".d.ts"
      ]
    },
    "import/resolver": {
      "<path>\\node_modules\\eslint-import-resolver-node\\index.js": {
        "extensions": [
          ".js",
          ".jsx",
          ".ts",
          ".tsx"
        ]
      },
      "<path>\\\node_modules\\eslint-import-resolver-typescript\\lib\\index.cjs": {
        "alwaysTryTypes": true
      }
    }
  },
```

**Comment by sweepline** [comment]
I need a minimal reproducible example. This supposedly works fine in my example here.
![image](https://github.com/sweepline/eslint-plugin-unused-imports/assets/2910395/5fd3319d-40a7-4f06-b82b-cffd845ad4fe)

I have not used nextJS. It looks very much like this plugin does not find `@typescript-eslint/eslint-plugin`. This means it falls back to the default eslint-rules which does not parse types correctly.

**Comment by julpat** [comment]
I had a similar issue and installing `@typescript-eslint/eslint-plugin` solved that 👍
